function/spring-computer-vision-function/README.adoc

```diff
@@ -0,0 +1,216 @@
+= Computer Vision Functions
+
+This module provides functional interface to perform common Computer Vision tasks such as Image Classification, Object Detection, Instance and Semantic Segmentation, Pose Estimation an more.
+
+It leverages the https://docs.djl.ai/index.html[Deep Java Library] (DJL) to enable Java developers to harness the power of deep learning.
+DJL serves as a bridge between the rich ecosystem of Java programming and the cutting-edge capabilities of deep learning.
+DJL provides integration with popular deep learning frameworks like `TensorFlow`, `PyTorch`, and `MXNet`, as well as support for a variety of pre-trained models using `ONNX Runtime`.
+
+== Beans for injection
+
+This module exposes auto-configurations for the following beans:
+
+* `Function<Message<byte[]>, Message<byte[]>> objectDetection` - Offering `Object Detection` for finding all instances of objects from a known set of categories in an image and `Instance Segmentation` for finding all instances of objects from a known set of categories in an image and drawing a mask on each instance.
+* `Function<Message<byte[]>, Message<byte[]>> imageClassifications` - The `Image Classification` task assigns a label to an image from a set of categories.
+* `Function<Message<byte[]>, Message<byte[]>> semanticSegmentation` - `Semantic Segmentation` refers to the task of detecting objects of various classes at pixel level.
+It colors the pixels based on the objects detected in that space.
+* `Function<Message<byte[]>, Message<byte[]>> poseEstimation` - `Pose Estimation` refers to the task of detecting human figures in images and videos, and estimating the pose of the bodies.
```

Comment:
I'm not sure where my comment is lost, but I'll repeat it.

It is better for end-user just to auto-wire a single function and leave the rest of hard logic up to internal routing.
You still can have those `Function` instances internally, but better to just expose only a single function for auto-configuration and so.

I would go with Spring Integration router, but probably a simple `switch` by some configuration property would be enough.

The point is that we pursue here a microservices architecture therefore a single auto-configured service from this module is what everyone would expect in their projects.

Any arguments that we have to leave it as a number of functions ?

Or... We might go with all those beans for functions since I see they are conditional, but let's see if we can make them with the same bean name in the end! So, target project would benefit from the same `@Autowire` without any paradox of choice for name.